blueocean-dashboard/src/main/js/PipelineRoutes.jsx

```diff
@@ -15,8 +15,84 @@ import {
     RunDetailsTests,
 } from './components';
 
+/**
+ * gets the background element used for the modal underlay
+ */
+function getBackgroundElement() {
+    return document.getElementById('modal-snap-background');
+}
+
+/**
+ * Cleans up the copied HTML to remove IDs, react root attributes
+ * and other non-visible elements.
+ */
+function cleanupCopy(el) {
```

Comment:
having duplicate IDs is bad, also don't want react to pick up anything (like the text node comments) erroneously